### PR TITLE
add negative part number test case for split part cross db util

### LINKED ISSUE #7915: [CT-2724] [Feature] Add negative test case for the `split_part` macro
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

The issue described in https://github.com/dbt-labs/dbt-bigquery/issues/615 is more general than dbt-bigquery, and the best way to support it is by adding a test cases for a negative index within dbt-core. 

### Describe alternatives you've considered

An option:
- each adapter maintainer is completely on their own to create test cases that include negative indices

This doesn't sound good for anyone, so let's go with Plan A!

### Who will this benefit?

This will benefit both adapter maintainers and downstream users of the `split_part` macro.

### Are you interested in contributing this feature?

🦞 is on it

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dave-connors-3**
resolves #7915

See also:
- https://github.com/dbt-labs/dbt-bigquery/issues/615
- https://github.com/dbt-labs/dbt-spark/issues/688

### Description

Adds test case with negative argument for `part_number` in `dbt.split_part` to ensure macro has intended behavior

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: .changes/unreleased/Fixes-20230320-153225.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+kind: Fixes
```
Comment (dbeatty10) [inline .changes/unreleased/Fixes-20230320-153225.yaml:1]: https://github.com/dbt-labs/dbt-bigquery/issues/615 is currently labeled as a feature / enhancement. If so, you'd want to update the `kind` accordingly.

Do you think it should be considered a bug instead?

**Review by dbeatty10 — CHANGES_REQUESTED** [review]
Could you check out the changie entry?

The new test case looks great!

File: .changes/unreleased/Fixes-20230320-153225.yaml (hunk 1 above)
Comment (mikealfare) [reply .changes/unreleased/Fixes-20230320-153225.yaml:1]: If you do change the `kind`, make sure that the `custom` area is updated accordingly. I believe bugs take an `Issue` whereas features take a `PR`. But check either way (which you could do by just generating a new changie log of the corresponding type).

File: .changes/unreleased/Fixes-20230320-153225.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+body: add negative part_number arg for split part macro
+time: 2023-03-20T15:32:25.5932-05:00
+custom:
+  Author: dave-connors-3
+  Issue: "615"
```
Comment (dbeatty10) [inline .changes/unreleased/Fixes-20230320-153225.yaml:6]: I just created #7915 so updating the changelog entry accordingly:
```suggestion
  Issue: "7915"
```

**Review by gshank — APPROVED** [review]
